### ISSUE TITLE
Comparison with ES7 / Babel / regenerator

### ISSUE THREAD

**Issue by dandv** [yortus/asyncawait#31]
http://pouchdb.com/2015/03/05/taming-the-async-beast-with-es7.html

**Comment by yortus** [comment]
Yep good idea, I'd like to add this when I get a chance.

**Comment by mrmayfield** [comment]
I think it's a great idea. I have a lot of proficiency with Babel but am hoping to use this in a side project, a quick overview of differences would help a lot.

**Comment by stanleyxu2005** [comment]
What will be the relationship of ES7. As the Node 7.x branch supports await/async natively already, I'd curious whether I should upgrade to Node 7.x with/without awaitasync.

**Comment by yortus** [comment]
@stanleyxu2005 once you can use native async/await, I'd recommend migrating away from this library to the native syntax. In most cases it should be as simple as removing the extra parentheses (unless you make use of deep `await`s, which aren't supported in the native syntax).

As for performance, I doubt it would matter much in real-world scenarios. Fibers (used by this lib) were much faster than ES6 generators in tight loops in V8, because V8 did not optimise generator functions. I'm not sure if that's still the case, or whether it also applies to V8's async/await. Either way any perf difference is likely to go away with ongoing V8 optimisation.

**Comment by allenhwkim** [comment]
@yortus I agree with @stanleyxu2005 await(function() ...) is better than the native syntax await function()....

**Comment by stanleyxu2005** [comment]
@allenhwkim  I think the developers can consider detect native async/await support and switch to native support (maybe for better performance) or use wrapper for old node environments. There is nothing we can do to change the async/await syntax but live with the fact. I'm now okay with it, only thing is still need more experience on troubleshooting unhandled rejections in PROD environment.
